src/RadhydroSimulation.hpp

```diff
@@ -1100,9 +1133,22 @@ auto RadhydroSimulation<problem_t>::advanceHydroAtLevel(amrex::MultiFab &state_o
 		amrex::iMultiFab redoFlag(grids[lev], dmap[lev], 1, 1);
 		redoFlag.setVal(quokka::redoFlag::none);
 
+		if constexpr (Physics_Traits<problem_t>::is_mhd_enabled) {
+			std::array<std::array<amrex::MultiFab, 2>, AMREX_SPACEDIM> emf_comps_ec;
+			for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+				emf_comps_ec[idim][0].define(
+				    amrex::convert(ba_cc, amrex::IntVect::TheDimensionVector(idim) + amrex::IntVect::TheDimensionVector((idim + 1) % 3)), dm, 1,
```

Comment:
I don't think this is correct. There should only be 1 emf MultiFab per spatial dimension. The emf multifabs should be identical to the `E` multifabs here: https://github.com/WeiqunZhang/amrex-devtests/blob/8ceb0e5e9ba61f553499bf19b21fcac21e25e7c7/edgefluxregister/main.cpp#L48